### PR TITLE
[LIVY-547][SERVER] Livy kills session after livy.server.session.timeout even if the session is active

### PR DESCRIPTION
**Pull request by shanyu**
## What changes were proposed in this pull request?
Add a new configuration:
  livy.server.session.timeout-check.skip-busy
To indicate whether or not to skip timeout check for a busy session. It defaults to false for 
backward compatibility.

https://issues.apache.org/jira/browse/LIVY-547

## How was this patch tested?

Manually tested the configuration.

### CODE REVIEW

**Comment by jerryshao** [comment]
@shanyu would you mind adding a unit test about it.

**Comment by shanyu** [comment]
Done. unit test added. But the CI build seems to failed with irrelevant errors.

File: server/src/test/scala/org/apache/livy/sessions/MockSession.scala [hunk 1]
```diff
@@ -31,7 +31,8 @@ class MockSession(id: Int, owner: String, conf: LivyConf, name: Option[String] =
 
   override def logLines(): IndexedSeq[String] = IndexedSeq()
 
-  override def state: SessionState = SessionState.Idle
+  var serverState: SessionState = SessionState.Idle
+  override def state: SessionState = { serverState }
```
Comment (jerryshao) [inline server/src/test/scala/org/apache/livy/sessions/MockSession.scala]: nit: no need to add `{ }` for one line method.

**Comment by jerryshao** [comment]
Reopen to trigger again.

**Comment by jerryshao** [comment]
@yiheng please also take a review. Thanks!

**Comment by yiheng** [comment]
LGTM.

**Comment by jerryshao** [comment]
Looks like there's some temporary network problems pulling dependences from maven repo. I'm going to merge it anyway, thanks for the contribution.

**Pull request closed without merging**